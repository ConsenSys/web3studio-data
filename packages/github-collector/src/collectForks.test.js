const { toArray, take } = require('rxjs/operators');
const collectRepos = require('./collectRepos');
const collectForks = require('./collectForks');
const mockGithub = require('../test/fixtures/mockGithub');
const TestTransport = require('../test/TestTransport');

describe('collectForks', () => {
  let source;
  let transport;

  afterAll(mockGithub.afterAll);
  beforeAll(mockGithub.beforeAll);

  beforeEach(() => {
    mockGithub.beforeEach();
    transport = new TestTransport();
    source = collectRepos().pipe(take(1));
  });

  it('Collects repo forks and passes as events', async () => {
    const results = await source
      .pipe(
        collectForks(transport),
        toArray()
      )
      .toPromise();

    expect(results.length).toEqual(7);

    expect(results).toMatchSnapshot();
  });

  it('Picks up from the last cursor', async () => {
    transport.mockLastEvent = { cursor: 'someCursor' };

    await source
      .pipe(
        collectForks(transport),
        toArray()
      )
      .toPromise();

    const requests = mockGithub.requests();

    // Second request is fetching the repo
    expect(requests[1].variables.cursor).toEqual(
      transport.mockLastEvent.cursor
    );
  });
});
