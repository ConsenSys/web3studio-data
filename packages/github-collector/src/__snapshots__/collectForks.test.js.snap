// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collectForks Collects repo forks and passes as events 1`] = `
Array [
  Object {
    "event": "fork",
    "meta": Object {
      "cursor": "Y3Vyc29yOnYyOpHOCdzZ0w==",
      "projects": Array [
        "web3studio-internal",
      ],
      "repo": "ConsenSys/web3studio-website",
      "user": Object {
        "id": "MDQ6VXNlcjE2MjQ=",
        "login": "breakpointer",
      },
    },
    "timestamp": "2019-01-13T05:15:28Z",
  },
  Object {
    "event": "fork",
    "meta": Object {
      "cursor": "Y3Vyc29yOnYyOpHOCf0sMA==",
      "projects": Array [
        "web3studio-internal",
      ],
      "repo": "ConsenSys/web3studio-website",
      "user": Object {},
    },
    "timestamp": "2019-01-25T17:33:38Z",
  },
  Object {
    "event": "fork",
    "meta": Object {
      "cursor": "Y3Vyc29yOnYyOpHOCjjwIg==",
      "projects": Array [
        "web3studio-internal",
      ],
      "repo": "ConsenSys/web3studio-website",
      "user": Object {
        "id": "MDQ6VXNlcjU3NzAwMDc=",
        "login": "barlock",
      },
    },
    "timestamp": "2019-02-19T15:56:50Z",
  },
  Object {
    "event": "fork",
    "meta": Object {
      "cursor": "Y3Vyc29yOnYyOpHOCjkIZA==",
      "projects": Array [
        "web3studio-internal",
      ],
      "repo": "ConsenSys/web3studio-website",
      "user": Object {
        "id": "MDQ6VXNlcjQyNjgzNjA0",
        "login": "hadasz",
      },
    },
    "timestamp": "2019-02-19T16:33:35Z",
  },
  Object {
    "event": "fork",
    "meta": Object {
      "cursor": "Y3Vyc29yOnYyOpHOCjzXCQ==",
      "projects": Array [
        "web3studio-internal",
      ],
      "repo": "ConsenSys/web3studio-website",
      "user": Object {
        "id": "MDQ6VXNlcjU2NzU=",
        "login": "rguiu",
      },
    },
    "timestamp": "2019-02-20T22:29:28Z",
  },
  Object {
    "event": "fork",
    "meta": Object {
      "cursor": "Y3Vyc29yOnYyOpHOCkvYAw==",
      "projects": Array [
        "web3studio-internal",
      ],
      "repo": "ConsenSys/web3studio-website",
      "user": Object {
        "id": "MDQ6VXNlcjM0NjE4NA==",
        "login": "KirkBallou",
      },
    },
    "timestamp": "2019-02-26T15:55:05Z",
  },
  Object {
    "event": "fork",
    "meta": Object {
      "cursor": "Y3Vyc29yOnYyOpHOCm-D6w==",
      "projects": Array [
        "web3studio-internal",
      ],
      "repo": "ConsenSys/web3studio-website",
      "user": Object {
        "id": "MDQ6VXNlcjE3NzgzODA=",
        "login": "scottrepreneur",
      },
    },
    "timestamp": "2019-03-11T20:40:05Z",
  },
]
`;
